### PR TITLE
Koonyu's homework for 2021-finalproject

### PR DESCRIPTION
**Pull request by Koonyu**
I have detailed my file
@MuZhou233

### CODE REVIEW

**Comment by Koonyu** [comment]
Homework finished
@MuZhou233

File: 2021-finalproject/Koonyu/check-environment/check-environment.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+if [ $ROS_MASTER_URI ];then
+	echo "[Check] environment variable... Pass"
+else
+	echo "[Check] environment variable... Failed"
+fi
+
+#check address validation " http://[[1-255].[0-255].[0-255].[1-255]]:[PORT] "
+#s/REGEX/SUBSTITUTION/
+#awk is used to analyze url which is valid or not
+if (export | grep ROS_MASTER_URI | sed -E "s/.*=//" | sed -E "s/\"//" | sed -E "s/\"//" | sed -E "s/http:\/\///" | sed -E "s/:.*//" | awk -F. '{if(NF==4 && $1>0 && $1<=255 && $2>=0 && $2<=255 && $3>=0 && $3<=255 && $4>0 && $4<255) exit 0; else exit 1}');then #check if this IP contents this model " [[1-255].[0-255].[0-255].[1-255]] "
```
Comment (MuZhou233) [inline 2021-finalproject/Koonyu/check-environment/check-environment.sh:12]: What will happen if the environment variable is empty

File: 2021-finalproject/README created by Koonyu.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+**2021/10/5
```
Comment (MuZhou233) [inline 2021-finalproject/README created by Koonyu.md:1]: Maybe duplicate with README?

File: 2021-finalproject/Koonyu/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+**2021/10/5
+# 学习总结
+
+
+**目前为止，我们除了学习C++的部分用法，还初步学习shell命令行、bash脚本、vim编辑器、ssh在数据整理的应用、密码学和git的基本用法。
```
Comment (MuZhou233) [inline 2021-finalproject/Koonyu/README.md:5]: Just in case, if you mean **bold** on this line, add two `*` at the end

**Review by MuZhou233 — APPROVED** [review]

File: 2021-finalproject/Koonyu/check-environment/check-environment.sh (hunk 1 above)
Comment (Koonyu) [reply 2021-finalproject/Koonyu/check-environment/check-environment.sh:12]: Before I didn't know that I could empty the environment variable with double quotation mark
Now ,  I can know that the result of emptying the environment variable is that all check failed

File: 2021-finalproject/Koonyu/README.md (hunk 3 above)
Comment (Koonyu) [reply 2021-finalproject/Koonyu/README.md:5]: Got it

File: 2021-finalproject/README created by Koonyu.md (hunk 2 above)
Comment (Koonyu) [reply 2021-finalproject/README created by Koonyu.md:1]: Got it

File: 2021-finalproject/Koonyu/check-environment/check-environment.sh [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+s=$(echo "obase=2;$(($g&$k))" | bc)
+t=$(echo "obase=2;$(($h&$l))" | bc)
+
+if (($m == $q && $n == $r && $o == $s && $p == $t));then
+	echo "[Check] same subnet...          Pass"
+else
+	echo "[Check] same subnet...          Failed"
+fi
+
+echo "All Check Passed"
```
Comment (MuZhou233) [inline 2021-finalproject/Koonyu/check-environment/check-environment.sh:63]: Through your screenshot, although all check failed, your script still print `All Check Passed`

**Review by MuZhou233 — CHANGES_REQUESTED** [review]

File: 2021-finalproject/Koonyu/check-environment/check-environment.sh (hunk 4 above)
Comment (Koonyu) [reply 2021-finalproject/Koonyu/check-environment/check-environment.sh:63]: I have rewrite script of bash . Three new screenshots can prove that it has no problem

**Comment by Koonyu** [comment]
I have rewritten script of bash . Three new screenshots can prove that it can solve problem that you had put forward before
Need to request review
@MuZhou233

**Comment by MuZhou233** [comment]
Pass the following check action first

**Comment by Koonyu** [comment]
Maybe I have known why my check often failed.
According the file ".github/workflows/2021-finalproject.yml"
 # Triggers the workflow on push or pull request events but only for the master branch
  pull_request:
    branches: [ master ]
However, I push my changes through patch-3 branch which I hadn't noticed for a long time before. I have no idea to deal with it beacuse you don't allow us make a new PR according requirement of homework
If I have other choice? @MuZhou233

**Comment by MuZhou233** [comment]
In addition, the path-check action will confirm **all** of your finally changes are in the right folder. Some of your current changes did not match that. check this https://github.com/gdut-dynamic-x/missing-semester/pull/50/files

**Comment by Koonyu** [comment]
@MuZhou233 
After renewing File changed, I haven't receive mail about "path-check failed". I can see a message "1 workflow awaiting approval'.
Maybe it has no problem. Need to request review

**Review by MuZhou233 — APPROVED** [review]

**Pull request merged**